packages/block-editor/src/components/responsive-block-control/index.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { uniqueId } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+import { Fragment, useState } from '@wordpress/element';
+
+import {
+	ToggleControl,
+} from '@wordpress/components';
+
+export function ResponsiveBlockControlLabel( { property, device } ) {
+	const accessibleLabel = sprintf( __( 'Controls the %s property for %s devices.' ), property, device );
+	const descId = uniqueId();
+	return (
+		<Fragment>
+			<span aria-describedby={ `rbc-desc-${ descId }` }>
+				{ device }
+			</span>
+			<span className="screen-reader-text" id={ `rbc-desc-${ descId }` }>{ accessibleLabel }</span>
+		</Fragment>
+	);
+}
+
+function ResponsiveBlockControl( props ) {
+	const { legend, property, toggleLabel, responsiveControlsActive = false, renderDefaultControl, defaultLabel = __( 'All' ), devices = [ __( 'Desktop' ), __( 'Tablet' ), __( 'Mobile' ) ], renderResponsiveControls } = props;
```

Comment:
I think the devices need an identifier like a slug that we have for colors. Right now they are only identified with a label that is tralatable we need a unique way to refer a device that does not change with the translation.

Reply:
Great point. I will sort this.

Reply:
Ok I've converted `devices` into an array of objects with the form:

```js
{
    id: 'small', // unique non-translated
    label: __('Small'), // purely for visual labels
}
```

Ditto for the default value of `All`.

This ensures we always have an unchanging value for the device.